app/models/concerns/reviewable_action_builder.rb

```diff
@@ -3,81 +3,71 @@
 module ReviewableActionBuilder
   extend ActiveSupport::Concern
 
+  attr_accessor :actions, :guardian, :action_args
+
   # Standard post-actions bundle. Consumers should use the returned
   # bundle value when adding post-focused actions.
   #
-  # @param actions [Reviewable::Actions] Actions instance to add the bundle to.
-  # @param guardian [Guardian] Guardian instance to check permissions.
-  #
   # @return [Reviewable::Actions::Bundle] The created post actions bundle.
-  def build_post_actions_bundle(actions, guardian)
-    bundle =
-      actions.add_bundle(
-        "#{id}-post-actions",
-        label: "reviewables.actions.post_actions.bundle_title",
-      )
-
-    # Always include the no-op action
-    build_action(actions, :no_action_post, bundle:)
+  def build_post_actions_bundle
+    bundle_actions = { no_action_post: {} }
+    if target_post
+      if target_post.trashed? && @guardian.can_recover_post?(target_post)
+        bundle_actions[:restore_post] = {}
+      end
 
-    return bundle unless target_post
+      if target_post.hidden?
+        bundle_actions[:unhide_post] = {} if !target_post.user_deleted?
+      else
+        bundle_actions[:hide_post] = {}
+      end
 
-    if target_post.trashed? && guardian.can_recover_post?(target_post)
-      build_action(actions, :restore_post, bundle:)
-    end
+      if @guardian.can_delete_post_or_topic?(target_post)
+        bundle_actions[:delete_post] = {}
+        bundle_actions[:delete_post_and_replies] = { confirm: true } if target_post.reply_count > 0
+      end
 
-    if target_post.hidden?
-      build_action(actions, :unhide_post, bundle:) if !target_post.user_deleted?
-    else
-      build_action(actions, :hide_post, bundle:)
-    end
+      bundle_actions[:edit_post] = { client_action: "edit" }
 
-    if guardian.can_delete_post_or_topic?(target_post)
-      build_action(actions, :delete_post, bundle:)
-      if target_post.reply_count > 0
-        build_action(actions, :delete_post_and_replies, bundle:, confirm: true)
+      if target_post.topic && !target_post.topic.private_message?
+        bundle_actions[:convert_to_pm] = {}
```

Comment:
Current guardian for this action is:
`    if topic && Guardian.new(performed_by).can_moderate?(topic)`